pkg/engine/template_generator.go

```diff
@@ -621,6 +621,14 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 		"IsCustomVNET": func() bool {
 			return cs.Properties.AreAgentProfilesCustomVNET()
 		},
+		"GetEscapedEnvironmentJSON": func() string {
+			customEnvironmentJSON, _ := cs.Properties.GetCustomEnvironmentJSON(false)
+			customEnvironmentJSON = strings.Replace(customEnvironmentJSON, `"`, `""`, -1)
+			return customEnvironmentJSON
+		},
+		"IsIdentitySystemADFS": func() bool {
+			return cs.Properties.IsAzureStackCloud() && cs.Properties.CustomCloudProfile.IdentitySystem == api.ADFSIdentitySystem
```

Comment:
Should we nil check `CustomCloudProfile` for Azure deployments?

Reply:
That is what `IsAzureStackCloud()` does. 
```
func (p *Properties) IsAzureStackCloud() bool {
	return p.CustomCloudProfile != nil
}
```
Would it be better to directly check if it is nil?

Reply:
no, it is fine, I forgot the function was already nit checking, thx